aten/src/ATen/native/cuda/ReflectionPadding1d.cu

```diff
@@ -0,0 +1,215 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/Utils.h>
+// keeping THC headers for atomicAdd
+#include <THC/THCAtomics.cuh>
+
+namespace at {
+namespace native {
+namespace {
+
+using at::cuda::detail::canUse32BitIndexMath;
+
+__device__
+inline void get_index_mapping(
+    int64_t input_w, int64_t output_w,
+    int64_t output_x,
+    int64_t pad_l,
+    int64_t & input_idx, int64_t & output_idx) {
```

Comment:
can you check if making these explicit returns slows things down (as a tuple or named struct)?  It's much clearer that way and it probably doesn't slow things down.